src/core/tiledmesh/qgstiledmeshboundingvolume.h

```diff
@@ -0,0 +1,184 @@
+/***************************************************************************
+                         qgstiledmeshboundingvolume.h
+                         --------------------
+    begin                : July 2023
+    copyright            : (C) 2023 by Nyall Dawson
+    email                : nyall dot dawson at gmail dot com
+ ******************************************************************
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSTILEDMESHBOUNDINGVOLUME_H
+#define QGSTILEDMESHBOUNDINGVOLUME_H
+
+#include "qgis_core.h"
+#include "qgis_sip.h"
+#include "qgis.h"
+#include "qgsbox3d.h"
+#include "qgssphere.h"
+#include "qgsorientedbox3d.h"
+#include "qgscoordinatetransform.h"
+
+/**
+ * \ingroup core
+ * \brief Abstract base class for bounding volumes for tiled mesh nodes.
+ *
+ * \since QGIS 3.34
+ */
+class CORE_EXPORT QgsAbstractTiledMeshNodeBoundingVolume
```

Comment:
Just a thought - the more I dealt with bounding volumes, the more I would prefer not to expose them, and keep all the various kinds of bounding volumes as an implementation detail / for debugging - and only expose the final axis-aligned bounding box (QgsBox3d), which would be used throughout QGIS code...

Reply:
Can we revisit this later? For the moment I'd like to expose the guts of these details as it makes debugging and testing much easier. And i think the final API considerations will become clearer the further we get here (and especially when we start implementing other non-cesium specifications like SLPK).

It's all marked as experimental api so we aren't limited by the api contract.

Reply:
Sounds good!